requirements/extra.txt

```diff
@@ -0,0 +1,15 @@
+# Extra dependencies required for development and tests
+
+-r base.txt
+
+# Pytest and the optional pytest coverage plugin for tests
+pytest
+pytest-cov
+
+# Optional dependencies for extra functionality
+sqlalchemy
+scikit-image
```

Comment:
Well, that would mean adding every dependency of scikit-image, not just networkx.
I'm not a fan of that since scikit-image will work around its issue the ways scipy does it. I talked to a core dev and he's waiting for my PR and I wanted to do it yesterday but...numpy and errors....
Here's the code from scipy:

    # We don't want to do that unconditionally, because we risk updating
    # an installed numpy which fails too often.  Just if it's not installed, we
    # may give it a try.  See gh-3379.
    build_requires = []
    try:
        import numpy
    except:
        build_requires = ['numpy>=1.6.2']